sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/SubResource.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing a sub-resource that contains only the ID.
+    /// </summary>
+    [ReferenceType]
+    public abstract class SubResource
```

Comment:
Missing corresponding SubResource.Serialization.cs file to serialize / deserialize this class